crypto/err/err_prn.c

```diff
@@ -44,6 +45,123 @@ void ERR_print_errors_cb(int (*cb) (const char *str, size_t len, void *u),
     }
 }
 
+/* auxiliary function for incrementally reporting texts via the error queue */
+static void put_error(int lib, const char *func, int reason,
+                      const char *file, int line)
+{
+    ERR_new();
+    ERR_set_debug(file, line, func);
+    ERR_set_error(lib, reason, NULL /* no data here, so fmt is NULL */);
+}
+
+#define TYPICAL_MAX_OUTPUT_BEFORE_DATA 100
+#define MAX_DATA_LEN (ERR_PRINT_BUF_SIZE - TYPICAL_MAX_OUTPUT_BEFORE_DATA)
+void ERR_add_error_txt(const char *separator, const char *txt)
+{
+    const char *file = NULL;
+    int line;
+    const char *func = NULL;
+    const char *data = NULL;
+    int flags;
+    unsigned long err = ERR_peek_last_error();
+
+    if (separator == NULL)
+        separator = "";
+    if (err == 0)
+        put_error(ERR_LIB_CMP, NULL, 0, "", 0);
```

Comment:
CMP specific errors...

Reply:
@levitte, the function itself can be more generally useful and belongs in the context of `crypto/err/`, that's why we moved it here. We just forgot to update this lib reference (and the one below) from `ERR_LIB_CMP` to `ERR_LIB_SYS`. We plan to fix this nit along with the next CMP chunk. 
@Akretsch, please include the commit I've just pushed to our `cmp-dev` branch.

Reply:
It turns out that the replacement here better is `ERR_LIB_NONE`.